pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -79,15 +79,13 @@ var (
 type Requirements struct {
 	// Requirements are layered with Labels and applied to every node.
 	Requirements []v1.NodeSelectorRequirement `json:"requirements,omitempty"`
-	// Ground truth for record keeping
-	Records map[string][]v1.NodeSelectorRequirement `json:"-"`
-	Allows  map[string]sets.Set                     `json:"-"`
+	Allows       map[string]sets.Set          `json:"-"`
```

Comment:
One trick you could use is to call this 
```suggestion
	requirements map[string]sets.Set          `json:"-"`
```
which implies that it's the same thing in a more compressed form.